src/applyMixin.js

```diff
@@ -1,7 +1,49 @@
+// Object.assign polyfill from
+// https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
+if (typeof Object.assign != 'function') {
+  (function () {
+    Object.assign = function (target) {
+      'use strict';
+      if (target === undefined || target === null) {
+        throw new TypeError('Cannot convert undefined or null to object');
+      }
+
+      var output = Object(target);
+      for (var index = 1; index < arguments.length; index++) {
+        var source = arguments[index];
+        if (source !== undefined && source !== null) {
+          for (var nextKey in source) {
+            if (source.hasOwnProperty(nextKey)) {
+              output[nextKey] = source[nextKey];
+            }
+          }
+        }
+      }
+      return output;
+    };
+  })();
+}
+
+/*
+ * Apply mixin (POJO) to ES6 class
+ * @param target {Class} - The class to decorate
+ * @param mixin {Object} - The object mixin to apply
+ * @return {Class} - Decorated class
+ */
 function applyMixin(target, mixin) {
   // apply mixin to the target class
   // and return the mixed class
   return Object.keys(mixin).reduce((targetClass, key) => {
+    // if the key is contextTypes
+    if (key === 'contextTypes') {
+      // merge the target and mixin contextTypes
+      const value = Object.assign({}, mixin[key], targetClass[key]);
+      // apply it
+      targetClass[key] = value;
```

Comment:
checkout if this change still does what you want
```javascript
targetClass[key] = { ...mixin[key], ...targetClass[key] };
```

When we run **npm run build** babel seems to cover our needs and we don't have to declare Object.assign ourselves.

In the mean time I'll be writing some tests to cover what you said.